src/Bicep.Core/Analyzers/Linter/Rules/WhatIfShortCircuitingRule.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Deployments.Core.Constants;
+using Azure.Deployments.Core.Definitions.Schema;
+using Azure.Deployments.Expression.Intermediate;
+using Azure.Deployments.Templates.Engines;
+using Azure.Deployments.Templates.ParsedEntities;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Intermediate;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.TypeSystem.Providers;
+using Microsoft.WindowsAzure.ResourceStack.Common.Collections;
+using Newtonsoft.Json;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class WhatIfShortCircuitingRule : LinterRuleBase
+    {
+        public new const string Code = "what-if-short-circuiting";
+
+        // IMPORTANT: Do not update this API version until the new one is confirmed to be deployed and available in ALL the clouds.
+        public const string NestedDeploymentResourceApiVersion = "2022-09-01";
```

Comment:
I think this was copied from somewhere else (`Azure.Bicep.Core.Emit.TemplateWriter`, maybe?). Since we want to make sure this isn't altered except as dictated by the above comment, could you move this to a constant that is used by both locations? `Azure.Bicep.Core.Emit.EmitConstants` would be a reasonable place to put it.